druid/src/env.rs

```diff
@@ -420,6 +420,12 @@ macro_rules! impl_value_type_borrowed {
                 Value::$var(self)
             }
         }
+
+        impl Into<Value> for &$ty {
```

Comment:
Just flagging this as interesting, I don't understand this enough to comment usefully.

Reply:
This is where we just about reach the limit of my experience and understanding of the system. Without that it's impossible to use `into` to turn a `Key<&str>` into a `KeyOrValue<&str>`, since you can't `into` a `&str` into a `Value`, and it's not possible to use `KeyOrValue<String>` since the `Key<T>` and `KeyOrValue<T>` have to match. I'm certain there are more elegant ways to do this, but I don't understand the system well enough to know how to implement them.

Reply:
I think you want a `Key<String>`, which can be borrowed as a `&str` with a lifetime derived from the value. This is similar to how you can use a `&str` to check if an item exists in a `HashSet<String>`?

Reply:
I started a topic in zulip to discuss implementation details.